### PR TITLE
PTL tests get skipped if no_mom_on_server is true when requirement decorator is set

### PR DESCRIPTION
**Pull request by shilpakodli**
#### Describe Bug or Feature
Scenario-1:
PTL multi-node tests get skipped if moms are on remote host.
For example: pbs_benchpress -t TestMultiNode -p servers=host1,moms=host2:host3 

Reason is, in requirement decorator, if there is no common host specified in servers and moms then it will check for no_mom_on_server flag is set or not. If it is not set then tests will be skipped.

Scenario-2:
Requirement decorator doesn't handle if we pass PBS confile in -p param "host@pbs_conf_file".
For example, if we pass -pserver=host1@/etc/pbs.conf2 then in requirement decorator, server will be considered as "host@/etc/pbs.conf2"

#### Describe Your Change
Do not skip tests if no_mom_on_server flag is not set in testcase/testsuite requirement decorator and in -p param in "pbs_benchpress" command.
Handle host names if pbs_conf file is passed in -p param list.

Modified code for no_comm_on_server as well as its behavior was also same as no_mom_on_server.

#### Attach Test and Valgrind Logs/Output
[test_execution_logs.txt](https://github.com/PBSPro/pbspro/files/3732718/test_execution_logs.txt)

### CODE REVIEW

File: test/fw/ptl/utils/plugins/ptl_test_runner.py [hunk 1]
```diff
@@ -734,16 +750,20 @@ def __are_requirements_matching(param_dic=None, test=None):
...
-            if eff_tc_req['no_mom_on_server']:
+            if eff_tc_req['no_mom_on_server'] or \
+               (set(_nomom) - set(_servers)) or \
+               _no_mom_on_server:
                 return False
         else:
-            if not eff_tc_req['no_mom_on_server']:
+            if not eff_tc_req['no_mom_on_server'] or \
+               not _no_mom_on_server \
+               and (set(_nomom) - set(_servers)):
```
[the remark this answers is not in the document]
Comment (shilpakodli) [reply test/fw/ptl/utils/plugins/ptl_test_runner.py]: Thanks Vincent for pointing. I made changes and tested. Please have review.

File: test/fw/ptl/utils/plugins/ptl_test_runner.py [hunk 2]
```diff
@@ -734,16 +750,19 @@ def __are_requirements_matching(param_dic=None, test=None):
             elif eff_tc_req['min_server_disk'] >= available_sys_disk:
                 return False
         if set(param_dic['moms']) & set(param_dic['servers']):
-            if eff_tc_req['no_mom_on_server']:
+            if eff_tc_req['no_mom_on_server'] or \
+               (set(_nomom) - set(_servers)) or \
+               _no_mom_on_server:
                 return False
         else:
```
Comment (vstumpf) [inline test/fw/ptl/utils/plugins/ptl_test_runner.py]: Now that I think about it, do we even need this else? Is it really necessary?

Comment (shilpakodli) [reply test/fw/ptl/utils/plugins/ptl_test_runner.py]: That's right Vincent. I thought the same. Removed else condition and it's working fine without else.

File: test/fw/ptl/utils/plugins/ptl_test_runner.py [hunk 3]
```diff
@@ -687,6 +695,11 @@ def __are_requirements_matching(param_dic=None, test=None):
         ts_requirements = {}
         tc_requirements = {}
         param_count = {}
+        _servers = param_dic['servers']
```
Comment (hirenvadalia) [inline test/fw/ptl/utils/plugins/ptl_test_runner.py]: small suggestion: create a set here only instead of every place we use these variables.

**Comment by shilpakodli** [comment]
Please find latest execution logs:
[TestPbsExecjobSuspendResume.txt](https://github.com/PBSPro/pbspro/files/3771275/TestPbsExecjobSuspendResume.txt)

**Comment by vstumpf** [comment]
Looks good, you have to resolve some merge conflicts though.

**Comment by shilpakodli** [comment]
@vstumpf I have rebased and squashed commits. Please take this PR forward.

**Pull request merged**